File: q2482/q2482.py
class Solution:
    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        r, c = len(grid), len(grid[0])
        rowOneCount = [0] * r
        rowZeroCount = [0] * r
        colOneCount = [0] * c
        colZeroCount = [0] * c

        for i in range(r): 
            for j in range(c): 
                if grid[i][j] == 1: 
                    rowOneCount[i] += 1
                    colOneCount[j] += 1
                else: 
                    rowZeroCount[i] += 1
                    colZeroCount[j] += 1
        
        diff = [[0] * c for _ in range(r)]
        for i in range(r): 
            for j in range(c): 
                diff[i][j] = rowOneCount[i] + colOneCount[j] - rowZeroCount[i] - colZeroCount[j]
            
        return diff
        

File: q2482/q2482.java
class Solution {
    public int[][] onesMinusZeros(int[][] grid) {
        int r = grid.length;
        int c = grid[0].length;

        int[] rowOneCount = new int[r];
        int[] rowZeroCount = new int[r];
        int[] colOneCount = new int[c];
        int[] colZeroCount = new int[c];

        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j] == 1) {
                    rowOneCount[i]++;
                    colOneCount[j]++;
                } else {
                    rowZeroCount[i]++;
                    colZeroCount[j]++;
                }
            }
        }

        int[][] diff = new int[r][c];
        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                diff[i][j] = rowOneCount[i] + colOneCount[j] - rowZeroCount[i] - colZeroCount[j];
            }
        }
        return diff;
    }
}